### ISSUE TITLE
Why write the struct 8 times ?

### ISSUE THREAD

**Issue by cassepipe** [gaultier/blog#14]
```nasm
 ; Send the handshake to the server: write(2).
  mov rax, SYSCALL_WRITE
  mov rdi, rdi
  lea rsi, [rsp]
  mov rdx, 12*8 
  syscall

  cmp rax, 12*8 ; Check that all bytes were written.
  jnz die
  ```
The struct is 12 bytes long ok but why multiply the number of bytes to write by 8 ?

Thanks for making that tutorial, it's great.

**Comment by gaultier** [comment]
Thanks for pointing this out, at first sight it seems indeed like a mistake. But then it is surprising that it works at all. Maybe the X Server is very lenient, I will look into it.

**Comment by gaultier** [comment]
You were right! It was a bad copy paste from my part, I must have copied it from the code creating the window (`x11_create_window`) which does extend the stack by `12*8` bytes, but the handshake is just 12 bytes  in our case (no authorization).

I fixed it in [e86b7dcd17983e8bd7b4f904eee207445ece4a28](https://github.com/gaultier/blog/commit/e86b7dcd17983e8bd7b4f904eee207445ece4a28) along with explanations.

Good catch! And thank you.

**Comment by cassepipe** [comment]
Wow, that was quick fix

Maybe is it worth adding that the reason the struct is 12 bytes and not 8, which is what you find if you add its member's sizes up, is because of alignment ? 

The fact that structs need to be "packed" in order to enforce that they are exactly the same size as the sum of their members could be an interesting point so that a reader who like memory accounting is not confused :)
